### PR TITLE
Overhauling Documentation

### LINKED ISSUE #110: Documentation Docstring overhaul
Make every viz function doc string link to geocat-examples script that uses it. 

Also @anissa111 mentioned a previous docstring overhaul in comp and uxarray. Find relevant PRs and issues to that for guidance. We want consistent docstring formatting across repositories. Several docstrings are rendering weirdly, will fix them all with this overhaul in one push.

### PR DESCRIPTION
**Pull request by jukent**
Want to match documentation in rest of GeoCAT ecosystem, and deal with inconsistent docstring formattings, removes package imports from functions.

Also does larger packaging structure changes

Adding examples to every function in a different pull request.
Closes #110

Possibly related to PRs from January https://github.com/NCAR/geocat-viz/pull/99 and https://github.com/NCAR/GeoCAT-examples/pull/493

### CODE REVIEW

**Comment by jukent** [comment]
Sorry for so so many commits. Each one I had a different error message, so that's good! Docs looks good now. Will wait for all tests to pass and then request reviews.

**Comment by jukent** [comment]
@erogluorhan took a look at this with me this morning and informally approved it. Waiting for a formal code review.

**Review by pilotchute — APPROVED** [review]

File: docs/conf.py [hunk 1]
```diff
@@ -10,100 +10,211 @@
...
+    'dask': ('https://docs.dask.org/en/latest/', None),
     'python': ('http://docs.python.org/3/', None),
-    'numpy': ('http://docs.scipy.org/doc/numpy/', None),
+    'numpy': ("https://numpy.org/doc/stable", None),
     'scipy': ('https://docs.scipy.org/doc/scipy/reference/', None),
     'xarray': ('http://xarray.pydata.org/en/stable/', None),
-    'matplotlib': ('https://matplotlib.org/stable/', None),
-    'cartopy': ('https://scitools.org.uk/cartopy/docs/latest/', None)
+    'pint': ('https://pint.readthedocs.io/en/stable/', None),
+    'cftime': ('https://unidata.github.io/cftime/', None),
```
Comment (kafitzgerald) [inline docs/conf.py:79]: Pretty sure we need to leave in matplotlib and cartopy here.  It looks like pint is useful, but not sure if we mention cftime at all.

File: docs/installation.rst [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
 
 The easiest way to install GeoCAT-viz is using
-`Conda <http://conda.pydata.org/docs/>`_::
+`Conda <http://conda.pydata.org/docs/>`__::
 
-    conda create -n geocat -c conda-forge -c ncar geocat-viz
+    conda create -n geocat -c conda-forge geocat-viz
 
-where :code:`geocat` is the name of a new conda environment, which can then be
+where ```geocat``` is the name of a new conda environment, which can then be
```
Comment (kafitzgerald) [inline docs/installation.rst:20]: extra set of `` around geocat

File: .readthedocs.yml [hunk 3]
```diff
@@ -9,16 +9,14 @@ build:
...
-
-python:
-   install:
-      - requirements: docs/requirements.txt
-      - method: pip
-        path: .
+  jobs:
+    post_create_environment:
+      - pip install geocat-f2py
+      - python -m pip install --no-cache-dir .
```
Comment (anissa111) [inline .readthedocs.yml:15]: This shouldn't be necessary for viz. This step is in comp to make sure that the f2py dependency is installed prior to the environment solving for the geocat-comp install.

File: docs/installation.rst [hunk 4]
```diff
@@ -93,18 +97,18 @@ Required dependencies for building and testing GeoCAT-viz
...
+    - `cartopy <https://scitools.org.uk/cartopy/docs/latest/>`__
 
 
 How to create a Conda environment for building GeoCAT-viz
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The GeoCAT-viz source code includes a conda environment definition file in
 the root directory that can be used to create a development environment containing
-all of the packages required to build GeoCAT-viz. The file :code:`conda_environment.yml`
+all of the packages required to build GeoCAT-viz. The file ``conda_environment.yml``
```
Comment (kafitzgerald) [inline docs/installation.rst:111]: pretty sure there's no longer a `conda_environment.yml`.  should this be `environment.yml`?

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    =src
+packages =
+    geocat
+    geocat.viz
+setup_requires =
+    setuptools_scm
+    setuptools
+    pip
+    geocat.f2py
+install_requires =
```
Comment (kafitzgerald) [inline setup.cfg:39]: are we missing some requirements here?

**Review by kafitzgerald — COMMENTED** [review]
Sorry this got scattered a bit between this and another comment or two.  There's a lot here (files / changes). 

Overall looks great though aside from the few things I pointed out.  Guessing those just didn't get caught when moving stuff over from comp.

File: .readthedocs.yml (hunk 3 above)
Comment (kafitzgerald) [reply .readthedocs.yml:15]: I was wondering about this as well.

File: docs/conf.py [hunk 6]
```diff
@@ -10,100 +10,211 @@
...
+current_year = datetime.datetime.now().year
+copyright = u'{}, University Corporation for Atmospheric Research'.format(
+    current_year)
+author = u'GeoCAT'
 
+# The version info for the project being documented
+version = gv.__version__.split("+")[0]
 # The full version, including alpha/beta/rc tags.
-release = read_version()
+release = gv.__version__
```
Comment (anissa111) [inline docs/conf.py:155]: This actually won't work in geocat-viz. In comp, this was actually a fairly recent change when we did the packaging and namespace rearrangement. Geocat-viz won't be able to do `gv.__version__`.

File: docs/user_api/index.rst [hunk 7]
```diff
@@ -1,45 +1,53 @@
...
 
 GeoCAT-viz Plotting Routines
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
+.. currentmodule:: geocat.viz.taylor
 .. autosummary::
    :toctree: ./generated/
 
-   geocat.viz.TaylorDiagram
+   geocat.viz.taylor.TaylorDiagram
```
Comment (anissa111) [inline docs/user_api/index.rst:15]: This is not wrong, but I figured I'd mention that if you explicitly list the current module, you don't actually need to link the full file path like

```python
.. currentmodule:: geocat.viz.taylor
.. autosummary::
   :toctree: ./generated/

   TaylorDiagram
```

File: docs/conf.py (hunk 6 above)
Comment (anissa111) [reply docs/conf.py:155]: Oh shoot! I didn't realize you'd also done the packaging structure changes. Could you please add that to the PR description?

File: src/geocat/viz/__init__.py [hunk 8]
```diff
@@ -1,2 +1,11 @@
 from .taylor import *
 from .util import *
+
+# get version from pyproject.toml
+from importlib.metadata import version as _version
+try:
+    __version__ = _version("geocat.comp")
```
Comment (anissa111) [inline src/geocat/viz/__init__.py:7]:
```suggestion
    __version__ = _version("geocat.viz")
```

File: setup.cfg [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+package_dir =
+    =src
+packages =
+    geocat
+    geocat.viz
+setup_requires =
+    setuptools_scm
+    setuptools
+    pip
+    geocat.f2py
```
Comment (anissa111) [inline setup.cfg:38]: Does viz actually need geocat-f2py?

File: setup.cfg [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    metpy
+    scipy
+    xarray
+    xskillscore
+    packaging
+tests_require =
+    pytest
+    netcdf4
+    geocat-datafiles
```
Comment (anissa111) [inline setup.cfg:52]: I think you've just pulled this from comp, but it should be worth going through the original requirements.txt and meta.yml to figure out which of these you actually need

File: .readthedocs.yml (hunk 3 above)
Comment (jukent) [reply .readthedocs.yml]: Thanks. Removed it!

File: docs/conf.py (hunk 1 above)
Comment (jukent) [reply docs/conf.py]: Done

File: setup.cfg (hunk 10 above)
Comment (jukent) [reply setup.cfg]: Oops done

**Comment by jukent** [comment]
Thanks for your reviews! There were a lot of changes, so easy to lose some things in the process.

@anissa111 can you say more about the feedstock?

**Comment by anissa111** [comment]
> Thanks for your reviews! There were a lot of changes, so easy to lose some things in the process.
> 
> @anissa111 can you say more about the feedstock?

@jukent Yeah! See this PR in the comp feedstock: https://github.com/conda-forge/geocat-comp-feedstock/pull/20/files

Also, the full changes for the packaging changes are here on geocat-comp: https://github.com/NCAR/geocat-comp/pull/389/files.

To do these changes, I'd also strongly suggest doing some local testing building the package to make sure the next release will still go smoothly.

File: docs/user_api/index.rst (hunk 7 above)
Comment (anissa111) [reply docs/user_api/index.rst:15]: Yeah, I saw that. I don't think you need to change it, but I figured I'd mention that this probably doesn't need to be like this.

**Comment by jukent** [comment]
Thanks @anissa111 I made the changes that I missed in those PRs It builds locally. Hopefully it still passes GH checks.

Well now I'm struggling to build the docs locally with a "Extension error:
Could not import extension myst_nb (exception: No module named 'myst_nb')
make: *** [html] Error 2" which I think drives from `docs/conf.py`. I can investigate tomorrow.

**Comment by anissa111** [comment]
Yeah, I think the packaging stuff is large enough you might want to pull it out of this PR. I still see some things I don't think are quite right.

I think the reason the checks were passing before you pulled more stuff over from comp is because `meta.yml` was overwriting any additions you'd made in `setup.cfg`, but I'm not 100% sure. We've actually entirely removed meta.yml in geocat-comp. 

But anyway, the packaging changes took a lot of fiddling around and testing over in comp if you want to talk about it more in depth

**Comment by jukent** [comment]
Yes I'd appreciate a walkthrough with you. Maybe during our coworking time tomorrow?

**Comment by anissa111** [comment]
> Yes I'd appreciate a walkthrough with you. Maybe during our coworking time tomorrow?

Yeah, that's a perfect time!

**Review by anissa111 — APPROVED** [review]
I think this is good now! The packaging stuff still makes me a little nervous, but I think it'll be okay. If not, we'll fix it!

**Review by kafitzgerald — APPROVED** [review]

**Pull request merged**